src/Moderation/Entity/Report.php

```diff
@@ -21,51 +21,125 @@
 
 namespace Friendica\Moderation\Entity;
 
+use Friendica\Moderation\Collection;
+
 /**
- * @property-read int            $id
- * @property-read int            $reporterId
- * @property-read int            $cid
- * @property-read string         $comment
- * @property-read string|null    $category
- * @property-read bool           $forward
- * @property-read array          $postUriIds
- * @property-read int            $uid
- * @property-read \DateTime|null $created
+ * @property-read int                     $id
+ * @property-read int                     $reporterCid
+ * @property-read int                     $cid
+ * @property-read int                     $gsid
+ * @property-read string                  $comment
+ * @property-read string                  $publicRemarks
+ * @property-read string                  $privateRemarks
+ * @property-read bool                    $forward
+ * @property-read int                     $category
+ * @property-read int                     $status
+ * @property-read int|null                $resolution
+ * @property-read int                     $reporterUid
+ * @property-read int|null                $lastEditorUid
+ * @property-read int|null                $assignedUid
+ * @property-read \DateTimeImmutable      $created
+ * @property-read \DateTimeImmutable|null $edited
+ * @property-read Collection\Report\Posts $posts
+ * @property-read Collection\Report\Rules $rules
  */
-class Report extends \Friendica\BaseEntity
+final class Report extends \Friendica\BaseEntity
 {
+	const CATEGORY_SPAM = 1;
+	const CATEGORY_ILLEGAL = 2;
+	const CATEGORY_SAFETY = 4;
+	const CATEGORY_UNWANTED = 8;
+	const CATEGORY_VIOLATION = 16;
+	const CATEGORY_OTHER = 32;
```

Comment:
Are you thinking about reporting multiple categories in the future? Independent from that I think that "other" should be a higher value like for example 128, so that we can add other categories in the future.

Reply:
In this case I would revert the order and add Other as first value. So every other report will be an addendum and 'Other' isn't at any position. Would be more intuitive I guess

Reply:
@annando Yes, that was the point of using round binary values for each category.

@nupplaphil I like your idea better in that context even if I’m not sure the category list will change in the future. Rule Violation is covering most of the custom violations admins could want.